src/merge_operator.rs

```diff
@@ -52,8 +52,16 @@ pub unsafe extern "C" fn full_merge_callback(
     let cb: &mut MergeOperatorCallback = &mut *(raw_cb as *mut MergeOperatorCallback);
     let operands = &mut MergeOperands::new(operands_list, operands_list_len, num_operands);
     let key: &[u8] = slice::from_raw_parts(raw_key as *const u8, key_len);
-    let oldval: &[u8] = slice::from_raw_parts(existing_value as *const u8, existing_value_len);
-    let mut result = (cb.merge_fn)(key, Some(oldval), operands);
+    let oldval = if existing_value.is_null() || existing_value_len == 0 {
+        None
+    } else {
+        assert!(existing_value_len <= isize::MAX as size_t);
```

Comment:
Just curious, why can't the maximum size of `existing_value_len` exceed 2 GiB?

Reply:
It should not in practice, I am just following Rust's requirements for from_raw_part

Reply:
https://doc.rust-lang.org/std/slice/fn.from_raw_parts.html#safety